src/cli/start.ts

```diff
@@ -103,23 +97,42 @@ ${formattedServices}`
     const configOverrides = this.buildConfigObject(flags)
     config.util.extendDeep(config, configOverrides)
 
-    if (flags.purge) {
-      sequelizeFactory()
-      await this.purge()
-    }
+    const backUpJob = new DbBackUpJob(ethFactory())
+    // An infinite loop which you can exit only with SIGINT/SIGKILL
+    while (true) {
+      let stopCallback: () => void = () => {
+        throw new Error('No stop callback was assigned!')
+      }
 
-    const app = await appFactory()
-    const port = config.get('port')
-    const server = app.listen(port)
+      // Run backup job
+      backUpJob.run()
 
-    process.on('unhandledRejection', err =>
-      logger.error(`Unhandled Rejection at: ${err}`)
-    )
+      // Promise that resolves when reset callback is called
+      const resetPromise = new Promise(resolve => {
+        appFactory({
+          appResetCallBack: () => resolve()
+        }).then(value => {
+          // Lets save the function that stops the app
+          stopCallback = value.stop
+        })
+      })
 
-    server.on('listening', () =>
-      this.log('Server started on port %d', port)
-    )
+      // Let see if we have to restart the app at some point most probably because
+      // reorgs outside of confirmation range.
+      await resetPromise
+
+      logger.warn('Reorg detected outside of confirmation range. Rebuilding the service\'s state!')
+      logger.info('Stopping service')
+      stopCallback()
+      backUpJob.stop()
+
+      // Restore DB from backup
+      await backUpJob.restoreDb(() => undefined)
```

Comment:
Why do you "ignore" the `errorCallback` here? And actually why there is even such a thing? Isn't rejected Promise enough?